lib/app.dart

```diff
@@ -14,10 +17,24 @@ class MyApp extends StatefulWidget {
 }
 
 class MyAppState extends State<MyApp> {
+  @override
+  void initState() {
+    NotificationService().requestPermission();
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
     return ResponsiveSizer(
       builder: (context, orientation, deviceType) {
+        AwesomeNotifications().setListeners(
+          onActionReceivedMethod: (ReceivedAction receivedAction) async {
+            if (receivedAction.channelKey == 'session_channel') {
+              FlutterConRouter.globalNavigatorKey.currentContext
+                  ?.pushReplacementNamed(FlutterConRouter.feedbackRoute);
+            }
+          },
+        );
```

Comment:
This can go into the `initState`

Reply:
You can have it like so in `initState`

```dart
AwesomeNotifications().setListeners(
      onActionReceivedMethod: NotificationService.onActionReceivedMethod,
      onNotificationCreatedMethod:
          NotificationService.onNotificationCreatedMethod,
      onNotificationDisplayedMethod:
          NotificationService.onNotificationDisplayedMethod,
      onDismissActionReceivedMethod:
          NotificationService.onDismissActionReceivedMethod,
    );
```

Reply:
In addition, once you move this into the init state, you can have your method defined in the `NotificationService` as follows:

```dart
static Future<void> onActionReceivedMethod(
    ReceivedAction receivedAction,
  ) async {
      Logger().f(receivedAction);
      switch (receivedAction.channelKey) {
        case receivedAction.channelKey:
          // Navigate with `go_router`
          break;
      }
}
```